packages/trackerless-network/src/logic/EntryPointDiscovery.ts

```diff
@@ -20,95 +19,48 @@ const parseEntryPointData = (dataEntries: DataEntry[]): PeerDescriptor[] => {
-        const factor = 2 ** i
-        const delay = baseDelay * factor
-        try {
-            await task()
-        } catch (e: any) {
-            logger.trace(`${description} failed, retrying in ${delay} ms`)
-        }
-        try { // Abort controller throws unexpected errors in destroy?
-            await wait(delay, abortSignal)
-        } catch (err) {
-            logger.trace(`${err}`)  // TODO Do we need logging?
-        }
-    }
-}
-
 const logger = new Logger(module)
 
 export const ENTRYPOINT_STORE_LIMIT = 8
-export const NETWORK_SPLIT_AVOIDANCE_LIMIT = 4
 
 interface EntryPointDiscoveryConfig {
     streamPartId: StreamPartID
     localPeerDescriptor: PeerDescriptor
-    layer1Node: Layer1Node
     fetchEntryPointData: (key: DhtAddress) => Promise<DataEntry[]>
     storeEntryPointData: (key: DhtAddress, data: Any) => Promise<PeerDescriptor[]>
     deleteEntryPointData: (key: DhtAddress) => Promise<void>
     storeInterval?: number
 }
 
 export class EntryPointDiscovery {
+
     private readonly abortController: AbortController
     private readonly config: EntryPointDiscoveryConfig
     private readonly storeInterval: number
-    private readonly networkSplitAvoidedNodes: Set<DhtAddress> = new Set()
     private isLocalNodeStoredAsEntryPoint = false
-    private networkSplitAvoidanceIsRunning = false
+    
     constructor(config: EntryPointDiscoveryConfig) {
         this.config = config
         this.abortController = new AbortController()
         this.storeInterval = this.config.storeInterval ?? 60000
     }
 
-    async discoverEntryPointsFromDht(knownEntryPointCount: number): Promise<FindEntryPointsResult> {
-        if (knownEntryPointCount > 0) {
-            return {
-                entryPointsFromDht: false,
-                discoveredEntryPoints: []
-            }
-        }
+    async discoverEntryPointsFromDht(excludeSet: Set<DhtAddress> = new Set()): Promise<PeerDescriptor[]> {
         const discoveredEntryPoints = await this.discoverEntryPoints()
-        if (discoveredEntryPoints.length === 0) {
-            discoveredEntryPoints.push(this.config.localPeerDescriptor)
-        }
-        return {
-            discoveredEntryPoints,
-            entryPointsFromDht: true
-        }
-    }
-
-    private async discoverEntryPoints(): Promise<PeerDescriptor[]> {
-        const dataKey = streamPartIdToDataKey(this.config.streamPartId)
-        const discoveredEntryPoints = await this.queryEntrypoints(dataKey)
-        const filtered = discoveredEntryPoints.filter((node) => 
-            !this.networkSplitAvoidedNodes.has(getNodeIdFromPeerDescriptor(node)))
-        // If all discovered entry points have previously been detected as offline, try again
+        const filtered = discoveredEntryPoints.filter((node) => !excludeSet.has(getNodeIdFromPeerDescriptor(node)))
         if (filtered.length > 0) {
             return filtered
+        // If all found entry points are filtered return the unfiltered list 
```

Comment:
Better to avoid this kind of surprise. If this logic is needed in caller, let's move it there (i.e. implement one of the "Future improvements")